spec/features/roles_spec.rb

```diff
@@ -0,0 +1,151 @@
+require 'spec_helper'
+
+feature Role do
+  let(:conference) { create(:conference) }
+  let(:role_names) { Role.all.each.map(&:name) }
+
+  shared_examples 'edit role successfully' do |role_name, by_role_name|
+    let!(:role) { Role.find_by(name: role_name, resource: conference) }
+    let!(:by_role) { Role.find_by(name: by_role_name, resource: conference) }
+    let!(:user_to_sign_in) { create(:user, role_ids: [by_role.id]) }
+
+    before :each do
+      sign_in user_to_sign_in
+      visit admin_conference_roles_path(conference.short_title)
+    end
+
+    scenario 'edits role', feature: true, js: true do
```

Comment:
This should be more specific and descriptive, for example something like
"edits role #{role_name}, when user has role #{by_role_name}"